### PR TITLE
[Editor] detect if files use JSX

### PR DESCRIPTION
**Pull request by DavideDeCenzo**
Associated Issue: #4694 

Here's the Pull Request Doc
https://devtools-html.github.io/debugger.html/CONTRIBUTING.html#pull-requests

Here's the Debugger's Testing doc
https://docs.google.com/document/d/1oBMRxV8A2ag2t22YsQOxTdEv0mXKzIg0tggJjRkU1S0/edit#.
Feel free to improve it!

### CODE REVIEW

File: src/components/Editor/index.js [hunk 1]
```diff
@@ -500,10 +502,13 @@ class Editor extends PureComponent<Props, State> {
     }
 
     if (selectedSource) {
+      const hasJsx = getSymbols(selectedSource);
```
Comment (bomsy) [inline src/components/Editor/index.js]: Does this mean that, if symbols exist for the selected source, then it has JSX? Not sure i understand this...

Comment (DavideDeCenzo) [reply src/components/Editor/index.js]: My bad you're right I got confused with another thing Thanks for the review!

**Comment by jasonLaster** [comment]
Hey @DavideDeCenzo this is getting closer and I bet this version works, which is cool.

Couple of architectural comments:
1) we shouldn't call `getSymbols` from the component, because calling it can be slow
2) we should import `getSymbols` from the `selectors` file and call it in connect.

```diff
diff --git a/src/components/Editor/index.js b/src/components/Editor/index.js
index aa3706e..4ce645d 100644
--- a/src/components/Editor/index.js
+++ b/src/components/Editor/index.js
@@ -22,7 +22,8 @@ import {
   getHitCountForSource,
   getCoverageEnabled,
   getSourceMetaData,
-  getConditionalPanelLine
+  getConditionalPanelLine,
+  getSymbols
 } from "../../selectors";
 
 // Redux actions
@@ -627,6 +628,7 @@ const mapStateToProps = state => {
   return {
     selectedLocation: getSelectedLocation(state),
     selectedSource,
+    symbols: getSymbols(state),
     searchOn: getActiveSearch(state) === "file",
     hitCount: getHitCountForSource(state, sourceId),
     selectedFrame: getSelectedFrame(state),
```

The difference is that here we get the symbols from the redux store and don't have to ask the worker for them. Sorry this is complicated, might help to watch some egghead videos :)

**Comment by DavideDeCenzo** [comment]
Ok I'll try to figure out these aspects , Thanks for the advices @jasonLaster

**Comment by jasonLaster** [comment]
This looks perfect. Will review carefully soon though!

File: src/utils/source.js [hunk 2]
```diff
@@ -235,7 +240,7 @@ function getMode(source: Source, sourceMetaData: SourceMetaDataType) {
 
   if (
     (url && url.match(/\.jsx$/i)) ||
-    (sourceMetaData && sourceMetaData.isReactComponent)
+    (sourceMetaData && (sourceMetaData.isReactComponent || symbols.hasJsx))
```
Comment (jasonLaster) [inline src/utils/source.js]: lets simplify this:

`sourceMetaData && (sourceMetaData.isReactComponent || symbols.hasJsx)` => `symbols.hasJsx`

Comment (DavideDeCenzo) [reply src/utils/source.js]: I've got "invalid left-hand side in arrow function parameters" and some token errors. I've tried to fix that but nothing

**Pull request closed without merging**